File: server-practice/message-server/hddc-domain/src/main/kotlin/cool/hdd/domain/post/entity/Post.kt
package cool.hdd.domain.post.entity

import cool.hdd.domain.post.value.*
import java.time.Instant

data class Post(
    val id: PostId,

    val title: Title,
    val content: Content,
    val author: Author,
    val status: PostStatus,

    val platformCategory: PlatformCategory,
    val product: Product,

    val viewCount: Int = 0,
    val createdAt: Instant,
    val updatedAt: Instant,
) {

    @JvmInline
    value class PostId(val value: Long)

    fun copy(
        title: Title,
        content: Content,
        status: PostStatus,
        platformCategory: PlatformCategory,
        product: Product,
        now: Instant
    ) = Post(
        id = this.id,
        title = title,
        content = content,
        author = this.author,
        status = status,
        platformCategory = platformCategory,
        product = product,
        viewCount = this.viewCount,
        createdAt = this.createdAt,
        updatedAt = now,
    )

    companion object {
        fun postIdInit() = PostId(0)
    }
}

File: server-practice/py-server/py_domain/post/entity/post.py
from dataclasses import dataclass
from typing import Optional
from datetime import datetime
from py_domain.post.value.title import Title
from py_domain.post.value.content import Content

@dataclass(frozen=True)
class PostId:
    value: int

@dataclass
class Post:
    id: PostId
    title: Title
    content: Content
    created_at: datetime
    updated_at: datetime

    def update(self, title: Title, content: Content) -> 'Post':
        self.title = title
        self.content = content
        self.updated_at = datetime.now()
        return self

